Month1/Week2/Steam/Steam/Steam.cpp

```diff
@@ -0,0 +1,343 @@
+#include "Steam.h"
+#include <string>
+
+//-------------------------------------------------------------------------------------------
+//										CONSTRUCTOR
+//-------------------------------------------------------------------------------------------
+Steam::Steam()
+{
+	Uncategorized = FCategory("Uncategorized");
+	ActiveCommand = MenuCommand::MainMenu;
+}
+
+//-------------------------------------------------------------------------------------------
+//										RUNNING APP
+//-------------------------------------------------------------------------------------------
+void Steam::RunApp()
+{
+	std::cout << "Welcome to Steam!" << std::endl;
+	while (ActiveCommand != MenuCommand::Exit)
+	{
+		switch (ActiveCommand)
+		{
+			case MenuCommand::MainMenu:
+				MainMenu();
+				break;
+
+			case MenuCommand::AddGame:
+				AddGame();
+				break;
+
+			case MenuCommand::ManageCategories:
+				ManageCategories();
+				break;
+
+			case MenuCommand::BrowseGames:
+				BrowseGames();
+				break;
+		}
+	}
+	system("CLS");
+
+	std::cout << "Thanks for using Steam! Please come back soon!" << std::endl;
+}
+
+//-------------------------------------------------------------------------------------------
+//										MAIN MENU
+//-------------------------------------------------------------------------------------------
+void Steam::MainMenu()
+{
+	do
+	{
+		system("CLS");
+
+		std::cout << "Welcome to Steam! What would you like to do?" << std::endl;
+		std::cout << "1 - Add Game" << std::endl;
+		std::cout << "2 - Manage Categories" << std::endl;
+		std::cout << "3 - Browse Games" << std::endl;
+		std::cout << "4 - Exit" << std::endl;
+
+		int Option;
+		std::cin >> Option;
+
+		while (!std::cin.good() || Option < 1 || Option > 4)
+		{
+			std::cin.clear();
+			std::cin.ignore(INT_MAX, '\n');
+			std::cout << "INVALID INPUT. Please try again." << std::endl;
+			std::cin >> Option;
+		}
+
+		ActiveCommand = (MenuCommand)Option;
+
+		std::cin.clear();
+		std::cin.ignore(INT_MAX, '\n');
+
+	} while (!std::cin.good());
+}
+
+//-------------------------------------------------------------------------------------------
+//										ADDING A GAME
+//-------------------------------------------------------------------------------------------
+void Steam::AddGame()
+{
+	int CategoryChosen = ChooseCategory();
+
+	FGame NewGame = CreateGame();
+
+	if (CategoryChosen == -1)
+	{
+		Uncategorized.AddGame(NewGame);
+	}
+	else
+	{
+		Categories.AddGameToCategory(CategoryChosen, NewGame);
+	}
+
+	ActiveCommand = MenuCommand::MainMenu;
+}
+
+//-------------------------------------------------------------------------------------------
+int Steam::ChooseCategory()
+{
+	system("CLS");
+
+	int CategoryChosen = -1;
```

Comment:
Choose Category is presenting problems when declared as const, since it is calling ValidateInt (reading up online this causes issues, but I still don't understand why). The error it presents is `'FSteam::ValidateInt': 2 overloads have no legal conversion for 'this' pointer`